File: backend/app/schemas/public.py
from pydantic import BaseModel


class PublicBookOut(BaseModel):
    id: int
    title: str
    subtitle: str | None = None
    authors: list[str] = []
    genre: str | None = None
    publication_year: int | None = None
    page_count: int | None = None
    cover_image_path: str | None = None
    shelf: str | None = None
    tags: list[str] = []
    status: str = "unread"
    rating: int | None = None


class PublicLibraryResponse(BaseModel):
    owner_name: str
    total_books: int
    books: list[PublicBookOut]


class ShareLinkConfig(BaseModel):
    share_slug: str | None
    is_public_shelf: bool
    share_url: str

File: frontend/src/types/public.ts
export interface PublicBook {
  id: number;
  title: string;
  subtitle: string | null;
  authors: string[];
  genre: string | null;
  publication_year: number | null;
  page_count: number | null;
  cover_image_path: string | null;
  shelf: string | null;
  tags: string[];
  status: string;
  rating: number | null;
}

export interface PublicLibraryResponse {
  owner_name: string;
  total_books: number;
  books: PublicBook[];
}

export interface ShareLinkConfig {
  share_slug: string | null;
  is_public_shelf: boolean;
  share_url: string;
}
